posthog/session_recordings/queries/session_recording_list_from_replay_summary.py

```diff
@@ -204,6 +205,40 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
         if not should_join_persons:
             return "", {}
         else:
+            """
+            for teams with large numbers of distinct ids it can be super slow to load
+            distinct ids, particularly when loading by person uuid
+            we're stuck needing data from A to reduce the data needed to load B
+            and A relying on B
+            so, for some teams, we artificially limit the number of distinct ids we load
+            by looking at the events table and only loading distinct ids that have events
+            in the last 30 days (or whatever the storage TTL is) and have a session_id
+            for teams that have a lot of distinct ids and need the optimisation there are probably
+            too many events to load all of them anyway so we restrict to pageviews
+            not everyone will have those, so we will manually add teams to this optimisation as necessary
+            person on events and person merging improvements should remove the need for this
+            before we've had to add too many teams
+            """
+            filter_by_events_presence = (
+                """
+                AND distinct_id in (
+                    SELECT distinct_id
+                    FROM events e
+                    WHERE team_id = %(team_id)s
+                        -- regardless of what other filters are applied
+                        -- limit by storage TTL
+                        AND e.timestamp >= %(clamped_to_storage_ttl)s
+                        -- make sure we don't get the occasional unexpected future event
+                        AND e.timestamp <= now()
+                        -- and then any time filter for the events query
+                        {events_timestamp_clause}
+                        and $session_id != ''
+                        and event = '$pageview') as load_fewer_rows
```

Comment:
I wonder if we could skip that so that as long as there are any events it'd work rather than just `$pageview`, for example for mobile that would be `$screen`, etc.

Reply:
switched to the default event approach - it's tidy and reduces copy-pasta errors in future - but if I have to add more than one or two teams to this then we should be prioritising removing the need for it

Reply:
this is very much a "what can I do for that customer today" fix